kroxylicious/src/main/java/io/kroxylicious/proxy/filter/KrpcFilterContext.java

```diff
@@ -43,6 +44,19 @@ public interface KrpcFilterContext {
      */
     void forwardRequest(ApiMessage request);
 
+    /**
+     * Send a message from a filter towards the broker, invoking upstream filters
+     * and being informed of the response via TODO.
+     * The response will pass through upstream filters prior to the handler being invoked.
+     * Response propagation will stop once the handler has completed,
+     * i.e. the downstream filters will not receive the response.
+     *
+     * @param apiVersion The version of the request to use
+     * @param request The request to send.
+     * @param <T> The type of the response
+     */
+    <T extends ApiMessage> ProxyFuture<T> sendRequest(short apiVersion, ApiMessage request);
```

Comment:
On reflection _every_ might be a bit much. But my thinking was that assuming once per channel would be wrong in the presence of a filter which was modifying `ApiVersions` responses for its own purposes (why might a filter do that? Consider the topic encryption filter without support for `sendRequest`: One way to avoid seeing topic ids would be to tell the client that the broker didn't support `Fetch` requests that used topics id).

So we have the choice of either:

* Doing the wrong thing and just storing and using the same `ApiVersions` everywhere (for every filter cinnected to the same broker).
* Trying to be clever enough in the core proxy code, e.g. by observing the `ApiVersions` responses after passing through each Filter and caching them, but re-using the same instance when they weren't mutated by a Filter
* Just letting/forcing each filter that wants to use `sendRequest` handle its own caching of versions for those APIs it wants to use.

Reply:
> Just letting/forcing each filter that wants to use sendRequest handle its own caching of versions for those APIs it wants to use.

Agreed offline that this should be the option to go with for now.